components/gitpod-db/src/typeorm/transactional-db-impl.ts

```diff
@@ -8,8 +8,15 @@ import { EntityManager } from "typeorm";
 import { TypeORM } from "./typeorm";
 import { inject, injectable, optional } from "inversify";
 
+type TransactionalCode<DB, R> = (db: DB, txCtx: TransactionalContext) => Promise<R>;
+
 export interface TransactionalDB<DB> {
-    transaction<R>(code: (db: DB) => Promise<R>): Promise<R>;
+    transaction<R>(txCtx: TransactionalContext | undefined, code: TransactionalCode<DB, R>): Promise<R>;
```

Comment:
I like the idea of the optional argument! Have you thought about making it the 2nd argument, to make it less interruptive? Also, it would be a the same position as for `code`.

Reply:
Jup, makes sense. :+1: